File: src/main/java/kitchenpos/menus/tobe/domain/MenuGroupRepository.java
package kitchenpos.menus.tobe.domain;

import kitchenpos.menus.tobe.domain.vo.MenuGroupId;

import java.util.List;
import java.util.Optional;

public interface MenuGroupRepository {
    MenuGroup save(MenuGroup menuGroup);

    Optional<MenuGroup> findById(MenuGroupId id);

    List<MenuGroup> findAll();
}

File: src/main/java/kitchenpos/menus/tobe/domain/MenuGroupRepository.kt
package kitchenpos.menus.tobe.domain

import java.util.*

fun MenuGroupRepository.findByIdOrNull(menuGroupId: UUID): MenuGroup? =
    findById(menuGroupId).orElse(null)

interface MenuGroupRepository {
    fun findById(menuGroupId: UUID): Optional<MenuGroup>

    fun save(menuGroup: MenuGroup): MenuGroup
}
